invenio_communities/assets/semantic-ui/js/invenio_communities/members/searchBarWithFilters.js

```diff
@@ -0,0 +1,244 @@
+    currentQueryState.filters.push(data.value);
+    updateQueryState(currentQueryState);
+  };
+
+  render() {
+    const { filterKey, filterValues, filterLabel, ...uiProps } = this.props;
+    const options = filterValues.map((filterValue) => ({
+      key: filterValue.key,
+      text: filterValue.label,
+      value: [filterKey, filterValue.key],
+    }));
+    return (
+      <Dropdown
+        {...uiProps}
+        icon={null}
+        trigger={triggerButton(filterLabel)}
+        options={options}
+        onChange={this.onChangeFilter}
+        selectOnBlur={false}
+      />
+    );
+  }
+}
+
+DropdownFilter.propTypes = {
+  updateQueryState: PropTypes.object.isRequired,
+  currentQueryState: PropTypes.func.isRequired,
+  filterKey: PropTypes.string.isRequired,
+  filterValues: PropTypes.array.isRequired,
+  filterLabel: PropTypes.string.isRequired,
+};
+
+class DropdownSort extends Component {
+  onChangeSort = (e, data) => {
+    const { updateQueryState, currentQueryState } = this.props;
+    currentQueryState.sortBy = data.value;
+    updateQueryState(currentQueryState);
+  };
+
+  render() {
+    const { filterKey, filterValues, filterLabel, ...uiProps } = this.props;
+    const options = filterValues.map((filterValue) => ({
+      key: filterValue.sortBy,
+      text: filterValue.text,
+      value: filterValue.sortBy,
+    }));
+    return (
+      <Dropdown
+        {...uiProps}
+        icon={null}
+        trigger={triggerButton(filterLabel)}
+        options={options}
+        onChange={this.onChangeSort}
+        selectOnBlur={false}
+      />
+    );
+  }
+}
+
+DropdownSort.propTypes = {
+  updateQueryState: PropTypes.object.isRequired,
+  currentQueryState: PropTypes.func.isRequired,
+  filterKey: PropTypes.string.isRequired,
+  filterValues: PropTypes.array.isRequired,
+  filterLabel: PropTypes.string.isRequired,
+};
+
+class FilterLabel extends Component {
+  render() {
+    const { filter, currentQueryState, updateQueryState } = this.props;
+    const currentFilters = currentQueryState.filters;
+    const filterType = filter[0].charAt(0).toUpperCase() + filter[0].slice(1);
+    const filterValue = filter[1].charAt(0).toUpperCase() + filter[1].slice(1);
+    return (
+      <Label className="ml-15 mt-15">
+        {`${filterType}: ${filterValue}`}
+        <Icon
+          onClick={() => {
+            const start = currentFilters.indexOf(filter);
+            currentFilters.splice(start, 1);
+            updateQueryState(currentQueryState);
+          }}
+          name="delete"
+        />
+      </Label>
+    );
+  }
+}
+
+FilterLabel.propTypes = {
+  updateQueryState: PropTypes.object.isRequired,
+  currentQueryState: PropTypes.func.isRequired,
+  filter: PropTypes.array.isRequired,
+};
+
+class SortLabel extends Component {
+  render() {
+    const { sort, currentQueryState, updateQueryState } = this.props;
+    const sortValue = sort.charAt(0).toUpperCase() + sort.slice(1);
```

Comment:
Is this to have the uppercase first letter? eventually you might have a `_` func for that

Reply:
Good point! Will be addresed in the next PR.

Reply:
addressed